src/unit_tests/agentlessd/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+# Generate agentlessd library
+file(GLOB agentlessd_files
+    ${SRC_FOLDER}/agentlessd/*.o)
+
+add_library(OS_AGENTLESSD_O STATIC ${agentlessd_files})
+
+set_source_files_properties(
+    ${agentlessd_files}
+    PROPERTIES
+    EXTERNAL_OBJECT true
+    GENERATED true
+)
+
+set_target_properties(
+    OS_AGENTLESSD_O
+    PROPERTIES
+    LINKER_LANGUAGE C
+)
+
+target_link_libraries(OS_AGENTLESSD_O ${WAZUHLIB} ${WAZUHEXT} -lpthread)
+
+#include wrappers
+include(${SRC_FOLDER}/unit_tests/wrappers/wazuh/shared/shared.cmake)
+
+list(APPEND agentlessd_names "test_agentlessd")
+#list(APPEND agentlessd_flags "-Wl,--wrap,Privsep_GetUser -Wl,--wrap,Privsep_GetGroup ${DEBUG_OP_WRAPPERS} \
+#                                -Wl,--wrap,StartMQ -Wl,--wrap,CreateThread ${STDIO_OP_WRAPPERS}")
+list(APPEND agentlessd_flags "${DEBUG_OP_WRAPPERS} ${STDIO_OP_WRAPPERS} \
+                            -Wl,--wrap,StartMQ -Wl,--wrap,CreateThread -Wl,--wrap,fprintf \
```

Comment:
You say me that do not use a  'global' CreateThread mock, instead should be use a local create_thread(), like even see in analysisd UT?

Reply:
I mean that we have added the wrappers for threads, but we aren't using them. 

> instead should be use a local create_thread()

If there is a global wrapper, we shouldn't create a local one. Even when we see it as an example.

Now I see that all this is because the global wrapper (in `wazuh/src/unit_tests/wrappers/wazuh/shared/pthreads_op_wrappers.c`) isn't really well defined

```
int __wrap_CreateThread(__attribute__((unused)) void * (*function_pointer)(void *),
                        __attribute__((unused))  void *data) {
    return 1;
}
```
So you can ignore this comment.
Thanks.